### DISCUSSION TITLE
카드 컴포넌트 리팩토링 도중 추상화 관련 의견 필요

### DISCUSSION THREAD

**Discussion by godhyeongman** [eRentronic/eRentronic-Client#56] [도움 필요]
## 상황
* 카드 컴포넌트 리팩토링 작업
* 컴포넌트의 사이즈가 다른 곳에 비해 커서 이른 레이아웃(디스플레이) 단위가 사용 될 때마다 나눔(템플릿 분류 컨벤션)
* 그런데 이는 템플릿 컨벤션임
* 카드 컴포넌트는 제품이라는 하나의 데이터에 의존함, 서버 컴포넌트가 맞음

## 진행 중 이던 작업 
![캡처](https://user-images.githubusercontent.com/88314186/197953735-8a8271c9-cc94-4b4c-ab69-154ca2f5538c.JPG)

현재 추상화는 거의 완료 되었는데 안하는게 나을것 같다는 생각이 들었습니다. JS 생각이 궁금하네요

**Comment by herrakam** [comment]
저는 추상화를 하는게 좋을거 같다는 생각이 들어요 일단 카드는 확실한 서버 컴포넌트이고, ui를 기준으로 분리하는건 템플릿에 관련된 컨벤션인게 사실입니다. 그런데 카드가 보여주는 데이터의 종류가 매우 많다 보니 추상화를 하지 않고 그대로 두기에는 컴포넌트의 크기가 너무 커지는 것 같습니다. 컴포넌트의 종류와 상관 없이 컴포넌트가 크기가 너무 커진다면 분리하는게 맞다는 생각이고, ui 기준 분리는 템플릿 컨벤션이라고 해도 다른 컴포넌트들에서도 적용시키기에 좋은 분리 기준 같아요!

**Reply by godhyeongman** [reply]
더 고민해보니 화면 에서 무언가 잘 못 되었을 때는 확실히 잘못된 점을 추상화된 이름으로 빠르게 찾아서 문제를 파악하는데 더 도움이 될 것 같네요.

이대로 진행해도 좋을것 같습니다. 단순 세부 마크업 역할만 부여하려니 뭔가 연관된 역할이 없나 고민이 되었던것 같아요. 아니면 카드 컴포넌트를 잘못 설계한것인가? 월요일에 나머지 정할것 정해서 공지에 올리겠습니다.
